metrics-exporter-http/src/lib.rs

```diff
@@ -27,21 +26,29 @@ pub struct HttpExporter<C, R> {
     address: SocketAddr,
 }
 
+/// The arguments used to construct a `HttpExporter` instance.
+pub struct HttpExporterArgs<C, R> {
+    /// The handle used to acquire metrics.
+    pub controller: C,
+    /// The `Recorder` which formats the collected metrics.
+    pub recorder: R,
+    /// The address to which the http server will be bound.
+    pub address: SocketAddr,
+}
+
 impl<C, R> HttpExporter<C, R>
 where
-    C: AsyncSnapshotProvider + Clone + Send + 'static,
-    C::SnapshotFuture: Send + 'static,
-    C::SnapshotError: Error + Send + Sync + 'static,
-    R: Recorder + Clone + Into<String> + Send + 'static,
+    C: Record + Clone + Send + 'static,
+    R: Recorder + Clone + Export<String> + Send + 'static,
```

Comment:
Don't think Clone is needed.

Reply:
In this case `Clone` actually means duplicate the configuration which seems to be needed for futures. This is being addressed by #30 since the `Clone` trait is confusing.